### PR TITLE
fix: Allow to enable/disable save-for-later feature

### LINKED ISSUE #7337: Handle selective cart error failures gracefully.
Currently if you don't have salective cart extension installed on backend and still have it enabled on frontend you will see infinite loop of requests for the cart. We shouldn't requests all the time if we get `notFound` errors.

### LINKED ISSUE #7350: Save for later action always displayed in cart details component
After save for later deprecations you can't disable selective cart feature even if you don't have selective cart extension installed.

### PR DESCRIPTION
**Pull request by marlass**
Closes #7350 closes #7337

### CODE REVIEW

File: projects/core/src/cart/services/cart-config.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import { CartConfig, CartConfigService } from '@spartacus/core';
+
+describe('CartConfigService', () => {
+  let mockCartConfig: CartConfig;
+  let service: CartConfigService;
+
+  beforeEach(() => {
+    // default spec cart config
+    mockCartConfig = {
+      cart: {
```
Comment (dunqan) [inline projects/core/src/cart/services/cart-config.service.spec.ts]: What about reusing already defined defaultCartConfig?
(if we plan to keep it ;-) )

File: projects/core/src/cart/cart.module.ts [hunk 2]
```diff
@@ -32,6 +34,7 @@ export class CartModule {
     return {
       ngModule: CartModule,
       providers: [
+        provideDefaultConfig(defaultCartConfig),
```
Comment (dunqan) [inline projects/core/src/cart/cart.module.ts]: Is it necessary to provide defaultConfig in this case?

File: projects/core/src/cart/services/cart-config.service.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import { CartConfig } from '../config/cart-config';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CartConfigService {
+  constructor(protected config: CartConfig) {}
+
+  isSelectiveCartEnabled(): boolean {
+    return this.config.cart.selectiveCart.enabled;
```
Comment (dunqan) [inline projects/core/src/cart/services/cart-config.service.ts]: Maybe, just for sake of reliability, we could return correct value even if we don't have default config provided?

Comment (marlass) [reply projects/core/src/cart/services/cart-config.service.ts]: Fixed

File: projects/core/src/cart/cart.module.ts (hunk 2 above)
Comment (marlass) [reply projects/core/src/cart/cart.module.ts]: default config removed. No need for it really in that simple case.

File: projects/core/src/cart/services/cart-config.service.spec.ts (hunk 1 above)
Comment (marlass) [reply projects/core/src/cart/services/cart-config.service.spec.ts]: Simplified ;)

**Review by dunqan — APPROVED** [review]

**Comment by marlass** [comment]
QA steps:
Try 3 combinations:
- backend with selective cart extension + feature in spartacus enabled
- backend without selective cart extension + feature in spartacus enabled
- any backend with feature in spartacus disabled

If the feature is disabled there shouldn't be any save for later buttons in cart. No cart not found errors.

It it is enabled and there is selective cart extension it should work correctly.

If it is enabled and there isn't a selective cart extension then it shouldn't trigger infinite loop.

**Pull request merged**